### PR TITLE
Simplify the AppDefinition rendering code

### PR DESCRIPTION
**Pull request by jsancio**

### CODE REVIEW

File: cosmos-test-common/src/test/scala/com/mesosphere/cosmos/render/MarathonLabelsSpec.scala [hunk 1]
```diff
@@ -1,144 +0,0 @@
...
-    assertResult(original.website)(result.website)
-    assertResult(original.framework.getOrElse(false))(result.framework.get)
-    assertResult(original.preInstallNotes)(result.preInstallNotes)
-    assertResult(original.postInstallNotes)(result.postInstallNotes)
-    assertResult(original.postUninstallNotes)(result.postUninstallNotes)
-    assertResult(original.licenses)(resultLicenses)
-    assertResult(original.images)(result.images.as[Option[universe.v3.model.Images]])
-  }
-
-}
```
Comment (jsancio) [inline cosmos-test-common/src/test/scala/com/mesosphere/cosmos/render/MarathonLabelsSpec.scala]: Moved the more interesting tests to `PackageDefinitionRendererSpec`.

File: cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/handler/PackageRenderHandlerSpec.scala [hunk 2]
```diff
@@ -24,20 +24,20 @@ class PackageRenderHandlerSpec extends FreeSpec {
   "PackageRenderHandler should" - {
 
     "succeed when attempting to render a service with a marathon template" in {
-      val expectedBody = RenderResponse(Json.fromFields(Seq(
+      val expectedBody = RenderResponse(JsonObject.fromIterable(List(
```
Comment (larioj) [inline cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/handler/PackageRenderHandlerSpec.scala]: Why is this better?

File: cosmos-json/src/main/scala/com/mesosphere/universe/common/JsonUtil.scala [hunk 3]
```diff
@@ -1,7 +1,24 @@
...
+
+  def merge(target: JsonObject, fragment: JsonObject): JsonObject = {
+    fragment.toList.foldLeft(target) { (updatedTarget, fragmentEntry) =>
+      val (fragmentKey, fragmentValue) = fragmentEntry
+      val targetValueOpt = updatedTarget(fragmentKey)
+
+      val mergedValue = (targetValueOpt.flatMap(_.asObject), fragmentValue.asObject) match {
+        case (Some(targetObject), Some(fragmentObject)) =>
+          Json.fromJsonObject(merge(targetObject, fragmentObject))
+        case _ => fragmentValue
```
Comment (larioj) [inline cosmos-json/src/main/scala/com/mesosphere/universe/common/JsonUtil.scala]: What is the intended behavior when fragmentValue is not an object?

Comment (jsancio) [inline cosmos-json/src/main/scala/com/mesosphere/universe/common/JsonUtil.scala]: I am not 100% sure. This is a copy of the existing code. I am going to document the method and make sure that we have tests that cover the behaviour for the more interesting cases.

Comment (jsancio) [inline cosmos-json/src/main/scala/com/mesosphere/universe/common/JsonUtil.scala]: @larioj I added documentation and more test cases.

**Comment by cruhland** [comment]
good-refactor

^ that's how you know it's a good refactor!

**Comment by jsancio** [comment]
Thanks for the reivew @cruhland and @larioj

**Pull request merged**